python/ray/experimental/serve/handle.py

```diff
@@ -36,26 +49,37 @@ def remote(self, *args, **kwargs):
             raise RayServeException(
                 "handle.remote must be invoked with keyword arguments.")
 
-        # get slo_ms before enqueuing the query
-        request_slo_ms = kwargs.pop("slo_ms", None)
-        if request_slo_ms is not None:
-            try:
-                request_slo_ms = float(request_slo_ms)
-                if request_slo_ms < 0:
-                    raise ValueError(
-                        "Request SLO must be positive, it is {}".format(
-                            request_slo_ms))
-            except ValueError as e:
-                raise RayServeException(str(e))
+        # get request params defaults before enqueuing the query
+        default_kwargs = RequestParams.get_default_kwargs()
+        request_param_kwargs = {}
+        for k in default_kwargs.keys():
+            fixed_kwarg_name = self._fix_kwarg_name(k)
+            request_param_kwargs[fixed_kwarg_name] = kwargs.pop(
+                k, default_kwargs[k])
+
+        try:
+            # check if request_slo_ms specified is correct or not
+            slo_ms = request_param_kwargs["request_slo_ms"]
+            slo_ms = self._check_slo_ms(slo_ms)
+            request_param_kwargs["request_slo_ms"] = slo_ms
+        except ValueError as e:
+            raise RayServeException(str(e))
+
+        # create request parameters required for enqueuing the request
+        request_params = RequestParams(self.endpoint_name, TaskContext.Python,
```

Comment:
why do we need ]use the full default_kwargs here? isn't it already specified in RequestParams constructor?

Reply:
We don't need the value of default kwargs : but still the key of kwargs is needed to check if any one of the RequestParams args are specified while doing a remote call.
Note: This makes remote call powerful than http